playwright-e2e/tests/dsm/kitUploadFlow/saliva-kit-upload-flow.spec.ts

```diff
@@ -0,0 +1,180 @@
+      await kitsWithoutLabelPage.assertCreateLabelsBtn();
+      await kitsWithoutLabelPage.assertReloadKitListBtn();
+      await kitsWithoutLabelPage.assertTableHeader();
+      await kitsWithoutLabelPage.assertPageTitle();
+      await kitsWithoutLabelPage.search(KitsColumnsEnum.SHORT_ID, shortID);
+      shippingID = (await kitsWithoutLabelPage.getData(KitsColumnsEnum.SHIPPING_ID)).trim();
+
+      // final scan
+      const finalScanPage = await navigation.selectFromSamples<FinalScanPage>(SamplesNavEnum.FINAL_SCAN);
+      await finalScanPage.assertPageTitle();
+      await finalScanPage.fillScanPairs([kitLabel, shippingID]);
+      await finalScanPage.save();
+
+      // kits sent page
+      const kitsSentPage = await navigation.selectFromSamples<KitsSentPage>(SamplesNavEnum.SENT);
+      await kitsSentPage.waitForLoad();
+      await kitsSentPage.assertPageTitle();
+      await kitsSentPage.assertDisplayedKitTypes(expectedKitTypes);
+      await kitsSentPage.selectKitType(kitType);
+      await kitsSentPage.assertReloadKitListBtn();
+      await kitsSentPage.assertTableHeader();
+      await kitsSentPage.search(KitsColumnsEnum.MF_CODE, kitLabel);
+      await kitsSentPage.assertDisplayedRowsCount(1);
+
+      // kits received page
+      const kitsReceivedPage = await navigation.selectFromSamples<KitsReceivedPage>(SamplesNavEnum.RECEIVED);
+      await kitsReceivedPage.kitReceivedRequest(kitLabel);
+      await kitsReceivedPage.waitForLoad();
+      await kitsReceivedPage.selectKitType(kitType);
+      await kitsReceivedPage.assertPageTitle();
+      await kitsReceivedPage.assertDisplayedKitTypes(expectedKitTypes);
+      await kitsReceivedPage.assertReloadKitListBtn();
+      await kitsReceivedPage.assertTableHeader();
+      await kitsReceivedPage.search(KitsColumnsEnum.MF_CODE, kitLabel);
+      await kitsReceivedPage.assertDisplayedRowsCount(1);
+      const receivedDate = await kitsReceivedPage.getData(KitsColumnsEnum.RECEIVED);
+
+      // checks if the uploaded kit is displayed on the participant's page, in the sample information tab
+      await navigation.selectFromStudy<ParticipantListPage>(StudyNavEnum.PARTICIPANT_LIST);
+      await participantListPage.assertPageTitle();
+      const searchPanel = await participantListPage.filters.searchPanel;
+      await searchPanel.open();
+      await searchPanel.text('Short ID', {textValue: shortID});
+      await searchPanel.search();
+      await participantListTable.openParticipantPageAt(0);
+      await participantPage.assertPageTitle();
+      const sampleInformationTab = await participantPage.clickTab<SampleInformationTab>(TabEnum.SAMPLE_INFORMATION);
+      await sampleInformationTab.assertKitType(kitLabel, kitType)
+      await sampleInformationTab
+        .assertValue(kitLabel, {info: SampleInfoEnum.STATUS, value: SampleStatusEnum.RECEIVED})
+      await sampleInformationTab
+        .assertValue(kitLabel, {info: SampleInfoEnum.RECEIVED, value: receivedDate})
```

Comment:
Do you think we could use more contents checks/assertions after kit is received on sampleInformationTab or another page?

Reply:
I think this should be enough. If not, I can add. NP